dojson/contrib/marc21/utils.py

```diff
@@ -42,84 +175,40 @@ def create_record(marcxml, correct=False, keep_singletons=True):
+            record.append((tag, text))
 
     datafield_iterator = tree.iter(tag='{*}datafield')
     for datafield in datafield_iterator:
-        tag = datafield.attrib.get('tag', '!')  # .encode("UTF-8")
-        ind1 = datafield.attrib.get('ind1', '!')  # .encode("UTF-8")
-        ind2 = datafield.attrib.get('ind2', '!')  # .encode("UTF-8")
-        # ind1, ind2 = _wash_indicators(ind1, ind2)
-        if ind1 in ('', '_'):
-            ind1 = ' '
-        if ind2 in ('', '_'):
-            ind2 = ' '
-        subfields = []
+        tag = datafield.attrib.get('tag', '!')
+        ind1 = datafield.attrib.get('ind1', '!')
+        ind2 = datafield.attrib.get('ind2', '!')
+        if ind1 in ('', ):
+            ind1 = '_'
+        if ind2 in ('', ):
+            ind2 = '_'
+        ind1 = ind1.replace(' ', '_')
+        ind2 = ind2.replace(' ', '_')
+
+        fields = []
         subfield_iterator = datafield.iter(tag='{*}subfield')
         for subfield in subfield_iterator:
             code = subfield.attrib.get('code', '!')  # .encode("UTF-8")
-            text = subfield.text
-            if text is None:
-                text = ''
-            else:
-                text = text  # .encode("UTF-8")
+            text = subfield.text or ''
             if text or keep_singletons:
-                subfields.append((code, text))
-        if subfields or keep_singletons:
-            text = ''
-            field_position_global += 1
-            record.setdefault(tag, []).append((subfields, ind1, ind2, text,
-                                               field_position_global))
-
-    class SaveDict(dict):
-        __getitem__ = dict.get
-
-    def dict_extend_helper(d, key, value):
-        """Helper function.
-        If the key is present inside the dictionary it creates a list (if
-        not present) and extends it with the new value.
-        Almost as in `list.extend`
-        """
-        if key in d:
-            current_value = d.get(key)
-            if not isinstance(current_value, list):
-                current_value = [current_value]
-            current_value.append(value)
-            value = current_value
-        d[key] = value
-
-    rec_tree = SaveDict()
-
-    for key, values in iteritems(record):
-        if key < '010' and key.isdigit():
-            rec_tree[key] = [value[3] for value in values]
-        else:
-            for value in values:
-                field = SaveDict()
-                for subfield in value[0]:
-                    dict_extend_helper(field, subfield[0], subfield[1])
-                dict_extend_helper(
-                    rec_tree,
-                    (key + value[1] + value[2]).replace(' ', '_'), field)
-
-    return rec_tree
+                fields.append((code, text))
+
+        if fields or keep_singletons:
+            key = '{0}{1}{2}'.format(tag, ind1, ind2)
+            record.append((key, GroupableOrderedDict(fields)))
+
+    return OrderedDict(record)
```

Comment:
I'm not sure if it's going to work for repeatable fields with same indicators. IMHO we should use `GroupableOrderedDict`. If there is no test-case I will try to provide one.